diff --git a/CHANGELOG.md b/CHANGELOG.md
deleted file mode 100644
index 095ed92631ae57d9caac683759c9b38491c92677..0000000000000000000000000000000000000000
diff --git a/dist-cjs/index.d.ts b/dist-cjs/index.d.ts
index ec996d145b83f637e0d43ad0db1f0e3ab87efb70..b033a8df604153626c203f3c808267a8c9592923 100644
--- a/dist-cjs/index.d.ts
+++ b/dist-cjs/index.d.ts
@@ -407,6 +407,18 @@ export declare function DefaultStylePanelContent({ styles }: TLUiStylePanelConte
  */
 export declare const DefaultToolbar: NamedExoticComponent<DefaultToolbarProps>;
 
+/** patched export for custom toolbar */
+export declare const ToggleToolLockedButton: NamedExoticComponent<ToggleToolLockedButtonProps>;
+export declare const OverflowingToolbar: NamedExoticComponent<OverflowingToolbarProps>;
+export declare interface ToggleToolLockedButtonProps {
+    activeToolId: string;
+}
+export declare interface OverflowingToolbarProps {
+    children?: ReactNode;
+}
+export declare function MobileStylePanel(): JSX_2.Element;
+/** patched export for custom toolbar */
+
 /** @public @react */
 export declare function DefaultToolbarContent(): JSX_2.Element;
 
@@ -2243,6 +2255,7 @@ export declare interface TLUiToolItem<TranslationKey extends string = string, Ic
     meta?: {
         [key: string]: any;
     };
+    style?: React.CSSProperties;
 }
 
 /** @public */
diff --git a/dist-esm/index.mjs b/dist-esm/index.mjs
index 4dfb8bc221bed3552682312efc6779af5a653d57..4adab6b34d9468e6296669da363dfc5be0c03ee0 100644
--- a/dist-esm/index.mjs
+++ b/dist-esm/index.mjs
@@ -146,6 +146,9 @@ import {
 import {
   DefaultToolbar
 } from "./lib/ui/components/Toolbar/DefaultToolbar.mjs";
+import { OverflowingToolbar } from "./lib/ui/components/Toolbar/OverflowingToolbar.mjs";
+import { ToggleToolLockedButton } from "./lib/ui/components/Toolbar/ToggleToolLockedButton.mjs";
+import { MobileStylePanel } from "./lib/ui/components/MobileStylePanel.mjs";
 import {
   ArrowDownToolbarItem,
   ArrowLeftToolbarItem,
@@ -458,6 +461,7 @@ export {
   LineShapeUtil,
   LineToolbarItem,
   MiscMenuGroup,
+  MobileStylePanel,
   MoveToPageMenu,
   NoteShapeTool,
   NoteShapeUtil,
@@ -465,6 +469,7 @@ export {
   OfflineIndicator,
   OpacitySlider,
   OvalToolbarItem,
+  OverflowingToolbar,
   PORTRAIT_BREAKPOINT,
   PageItemInput,
   PageItemSubmenu,
@@ -550,6 +555,7 @@ export {
   ToggleReduceMotionItem,
   ToggleSnapModeItem,
   ToggleToolLockItem,
+  ToggleToolLockedButton,
   ToggleTransparentBgMenuItem,
   ToggleWrapModeItem,
   ToolbarItem,
diff --git a/dist-esm/lib/ui/components/primitives/Button/TldrawUiButtonIcon.mjs b/dist-esm/lib/ui/components/primitives/Button/TldrawUiButtonIcon.mjs
index fcb8268d733b28ab2d31a91e60818d46dbcfc3a2..812c7e77c5209abbd389f209ffae3dcb66c302c8 100644
--- a/dist-esm/lib/ui/components/primitives/Button/TldrawUiButtonIcon.mjs
+++ b/dist-esm/lib/ui/components/primitives/Button/TldrawUiButtonIcon.mjs
@@ -1,7 +1,7 @@
 import { jsx } from "react/jsx-runtime";
 import { TldrawUiIcon } from "../TldrawUiIcon.mjs";
-function TldrawUiButtonIcon({ icon, small, invertIcon }) {
-  return /* @__PURE__ */ jsx(TldrawUiIcon, { className: "tlui-button__icon", icon, small, invertIcon });
+function TldrawUiButtonIcon({ icon, small, invertIcon, style }) {
+  return /* @__PURE__ */ jsx(TldrawUiIcon, { className: "tlui-button__icon", icon, small, invertIcon, style });
 }
 export {
   TldrawUiButtonIcon
diff --git a/dist-esm/lib/ui/components/primitives/TldrawUiIcon.mjs b/dist-esm/lib/ui/components/primitives/TldrawUiIcon.mjs
index c588a438b131135eaa954247aa5aaf3048e28afd..fa7b127a874bfa41fb5290b41f9ddb2bb19fcb93 100644
--- a/dist-esm/lib/ui/components/primitives/TldrawUiIcon.mjs
+++ b/dist-esm/lib/ui/components/primitives/TldrawUiIcon.mjs
@@ -8,6 +8,7 @@ const TldrawUiIcon = memo(function TldrawUiIcon2({
   icon,
   color,
   className,
+  style,
   ...props
 }) {
   const assetUrls = useAssetUrls();
@@ -42,7 +43,8 @@ const TldrawUiIcon = memo(function TldrawUiIcon2({
       style: {
         color,
         mask: `url(${asset}) center 100% / 100% no-repeat`,
-        transform: invertIcon ? "scale(-1, 1)" : void 0
+        transform: invertIcon ? "scale(-1, 1)" : void 0,
+        ...style
       }
     }
   );
diff --git a/dist-esm/lib/ui/components/primitives/menus/TldrawUiMenuItem.mjs b/dist-esm/lib/ui/components/primitives/menus/TldrawUiMenuItem.mjs
index c05d70e3ddd80a9cfdbd0527b93cd6fcd01fed88..2de1239ff423486c73a85df4057948f666a5e2b5 100644
--- a/dist-esm/lib/ui/components/primitives/menus/TldrawUiMenuItem.mjs
+++ b/dist-esm/lib/ui/components/primitives/menus/TldrawUiMenuItem.mjs
@@ -23,7 +23,8 @@ function TldrawUiMenuItem({
   icon,
   onSelect,
   noClose,
-  isSelected
+  isSelected,
+  style
 }) {
   const { type: menuType, sourceId } = useTldrawUiMenuContext();
   const msg = useTranslation();
@@ -150,7 +151,7 @@ function TldrawUiMenuItem({
           },
           role: "radio",
           "aria-checked": isSelected ? "true" : "false",
-          children: /* @__PURE__ */ jsx(TldrawUiButtonIcon, { icon })
+          children: /* @__PURE__ */ jsx(TldrawUiButtonIcon, { icon, style })
         }
       );
     }
@@ -168,7 +169,7 @@ function TldrawUiMenuItem({
           role: "radio",
           "aria-checked": isSelected ? "true" : "false",
           "data-value": id,
-          children: /* @__PURE__ */ jsx(TldrawUiButtonIcon, { icon })
+          children: /* @__PURE__ */ jsx(TldrawUiButtonIcon, { icon, style })
         }
       ) });
     }
