diff --git a/CHANGELOG.md b/CHANGELOG.md
deleted file mode 100644
index 16e0590cc110366666eee3a0fa011d4ee9655fd1..0000000000000000000000000000000000000000
diff --git a/dist-cjs/index.d.ts b/dist-cjs/index.d.ts
index b1dfaf8b8445a6353327a07768e3a41377f834bf..ac3688d2bc193b6c52d2ba222104002a1e8e32bb 100644
--- a/dist-cjs/index.d.ts
+++ b/dist-cjs/index.d.ts
@@ -414,6 +414,18 @@ export declare function DefaultStylePanelContent({ styles }: TLUiStylePanelConte
  */
 export declare const DefaultToolbar: NamedExoticComponent<DefaultToolbarProps>;
 
+/** patched export for custom toolbar */
+export declare const ToggleToolLockedButton: NamedExoticComponent<ToggleToolLockedButtonProps>;
+export declare const OverflowingToolbar: NamedExoticComponent<OverflowingToolbarProps>;
+export declare interface ToggleToolLockedButtonProps {
+    activeToolId: string;
+}
+export declare interface OverflowingToolbarProps {
+    children?: ReactNode;
+}
+export declare function MobileStylePanel(): JSX_2.Element;
+/** patched export for custom toolbar */
+
 /** @public @react */
 export declare function DefaultToolbarContent(): JSX_2.Element;
 
@@ -1165,6 +1177,24 @@ export declare function RectangleToolbarItem(): JSX_2.Element;
  */
 export declare function removeFrame(editor: Editor, ids: TLShapeId[]): void;
 
+/** @public */
+// fixed in 3.0.0
+export declare function registerDefaultSideEffects(editor: Editor): (() => void)[];
+
+/** @public */
+// fixed in 3.0.0
+export declare function registerDefaultExternalContentHandlers(
+    editor: Editor,
+    {
+        maxImageDimension,
+        maxAssetSize,
+        acceptedImageMimeTypes,
+        acceptedVideoMimeTypes,
+    }: Required<TLExternalContentProps>,
+    { toasts, msg }: { toasts: TLUiToastsContextType; msg: ReturnType<typeof useTranslation> }
+): void;
+
+
 /** @public @react */
 export declare function RemoveFrameMenuItem(): JSX_2.Element | null;
 
@@ -2280,6 +2310,7 @@ export declare interface TLUiToolItem<TranslationKey extends string = string, Ic
     meta?: {
         [key: string]: any;
     };
+    style?: React.CSSProperties
 }
 
 /** @public */
diff --git a/dist-esm/index.mjs b/dist-esm/index.mjs
index e4d9298c6b2d649497490db730294d9ab4df1fef..a3821b15cad2d3fc217efcd35c8ab8c1d116b0d6 100644
--- a/dist-esm/index.mjs
+++ b/dist-esm/index.mjs
@@ -8,6 +8,8 @@ import { TldrawScribble } from "./lib/canvas/TldrawScribble.mjs";
 import { TldrawSelectionBackground } from "./lib/canvas/TldrawSelectionBackground.mjs";
 import { TldrawSelectionForeground } from "./lib/canvas/TldrawSelectionForeground.mjs";
 import { defaultBindingUtils } from "./lib/defaultBindingUtils.mjs";
+import { registerDefaultExternalContentHandlers } from "./lib/defaultExternalContentHandlers.mjs"; // fixed in 3.0.0
+import { registerDefaultSideEffects } from "./lib/defaultSideEffects.mjs"; // fixed in 3.0.0
 import { defaultShapeTools } from "./lib/defaultShapeTools.mjs";
 import { defaultShapeUtils } from "./lib/defaultShapeUtils.mjs";
 import { defaultTools } from "./lib/defaultTools.mjs";
@@ -138,6 +140,13 @@ import {
 import {
   DefaultToolbar
 } from "./lib/ui/components/Toolbar/DefaultToolbar.mjs";
+import {
+  ToggleToolLockedButton
+} from "./lib/ui/components/Toolbar/ToggleToolLockedButton.mjs";
+import {
+  OverflowingToolbar
+} from "./lib/ui/components/Toolbar/OverflowingToolbar.mjs";
+import { MobileStylePanel } from "./lib/ui/components/MobileStylePanel.mjs";
 import {
   ArrowDownToolbarItem,
   ArrowLeftToolbarItem,
@@ -441,6 +450,7 @@ export {
   LineShapeUtil,
   LineToolbarItem,
   MiscMenuGroup,
+  MobileStylePanel,
   MoveToPageMenu,
   NoteShapeTool,
   NoteShapeUtil,
@@ -448,6 +458,7 @@ export {
   OfflineIndicator,
   OpacitySlider,
   OvalToolbarItem,
+  OverflowingToolbar,
   PORTRAIT_BREAKPOINT,
   PageItemInput,
   PageItemSubmenu,
@@ -531,6 +542,7 @@ export {
   ToggleReduceMotionItem,
   ToggleSnapModeItem,
   ToggleToolLockItem,
+  ToggleToolLockedButton,
   ToggleTransparentBgMenuItem,
   ToggleWrapModeItem,
   ToolbarItem,
@@ -605,6 +617,8 @@ export {
   useToasts,
   useTools,
   useTranslation,
+  registerDefaultExternalContentHandlers, // fixed in 3.0.0
+  registerDefaultSideEffects, // fixed in 3.0.0
   useUiEvents
 };
 //# sourceMappingURL=index.mjs.map
diff --git a/dist-esm/lib/ui/components/Toolbar/OverflowingToolbar.mjs b/dist-esm/lib/ui/components/Toolbar/OverflowingToolbar.mjs
index 7a41748b09bf12d0c97d645df38897c9ef89f799..738836597d537a0ce067c0b8523cd923f9bde763 100644
--- a/dist-esm/lib/ui/components/Toolbar/OverflowingToolbar.mjs
+++ b/dist-esm/lib/ui/components/Toolbar/OverflowingToolbar.mjs
@@ -15,10 +15,15 @@ import {
   TldrawUiDropdownMenuTrigger
 } from "../primitives/TldrawUiDropdownMenu.mjs";
 import { TldrawUiMenuContextProvider } from "../primitives/menus/TldrawUiMenuContext.mjs";
+import { nanoid } from "nanoid";
+
+// query builder note
+// useSafeId() uses React 18's useId()
+// we only have access to 17.0.2
 const IsInOverflowContext = createContext(false);
 function OverflowingToolbar({ children }) {
   const editor = useEditor();
-  const id = useSafeId();
+  const id = "overflowing-toolbar";
   const breakpoint = useBreakpoint();
   const msg = useTranslation();
   const overflowIndex = Math.min(8, 5 + breakpoint);
diff --git a/dist-esm/lib/ui/components/primitives/Button/TldrawUiButtonIcon.mjs b/dist-esm/lib/ui/components/primitives/Button/TldrawUiButtonIcon.mjs
index fcb8268d733b28ab2d31a91e60818d46dbcfc3a2..5e2b869280fb094e6e5b1a1615d900ea7b87dd51 100644
--- a/dist-esm/lib/ui/components/primitives/Button/TldrawUiButtonIcon.mjs
+++ b/dist-esm/lib/ui/components/primitives/Button/TldrawUiButtonIcon.mjs
@@ -1,7 +1,7 @@
 import { jsx } from "react/jsx-runtime";
 import { TldrawUiIcon } from "../TldrawUiIcon.mjs";
-function TldrawUiButtonIcon({ icon, small, invertIcon }) {
-  return /* @__PURE__ */ jsx(TldrawUiIcon, { className: "tlui-button__icon", icon, small, invertIcon });
+function TldrawUiButtonIcon({ icon, small, invertIcon, style }) {
+  return /* @__PURE__ */ jsx(TldrawUiIcon, { className: "tlui-button__icon", icon, small, invertIcon, style });
 }
 export {
   TldrawUiButtonIcon
diff --git a/dist-esm/lib/ui/components/primitives/TldrawUiIcon.mjs b/dist-esm/lib/ui/components/primitives/TldrawUiIcon.mjs
index c588a438b131135eaa954247aa5aaf3048e28afd..64ddcf6991ccd9f6f63322f90b5661c5ad070457 100644
--- a/dist-esm/lib/ui/components/primitives/TldrawUiIcon.mjs
+++ b/dist-esm/lib/ui/components/primitives/TldrawUiIcon.mjs
@@ -8,6 +8,7 @@ const TldrawUiIcon = memo(function TldrawUiIcon2({
   icon,
   color,
   className,
+  style,
   ...props
 }) {
   const assetUrls = useAssetUrls();
@@ -42,7 +43,8 @@ const TldrawUiIcon = memo(function TldrawUiIcon2({
       style: {
         color,
         mask: `url(${asset}) center 100% / 100% no-repeat`,
-        transform: invertIcon ? "scale(-1, 1)" : void 0
+        transform: invertIcon ? "scale(-1, 1)" : void 0,
+        ...style
       }
     }
   );
diff --git a/dist-esm/lib/ui/components/primitives/menus/TldrawUiMenuItem.mjs b/dist-esm/lib/ui/components/primitives/menus/TldrawUiMenuItem.mjs
index d4533e5df517ae2905ae1543f9d90fdbecc9e922..49997e49ecbfde662ee7d95591e356f53aadec41 100644
--- a/dist-esm/lib/ui/components/primitives/menus/TldrawUiMenuItem.mjs
+++ b/dist-esm/lib/ui/components/primitives/menus/TldrawUiMenuItem.mjs
@@ -23,7 +23,8 @@ function TldrawUiMenuItem({
   icon,
   onSelect,
   noClose,
-  isSelected
+  isSelected,
+  style
 }) {
   const { type: menuType, sourceId } = useTldrawUiMenuContext();
   const msg = useTranslation();
@@ -153,7 +154,7 @@ function TldrawUiMenuItem({
           },
           role: "radio",
           "aria-checked": isSelected ? "true" : "false",
-          children: /* @__PURE__ */ jsx(TldrawUiButtonIcon, { icon })
+          children: /* @__PURE__ */ jsx(TldrawUiButtonIcon, { icon, style })
         }
       );
     }
@@ -171,7 +172,7 @@ function TldrawUiMenuItem({
           role: "radio",
           "aria-checked": isSelected ? "true" : "false",
           "data-value": id,
-          children: /* @__PURE__ */ jsx(TldrawUiButtonIcon, { icon })
+          children: /* @__PURE__ */ jsx(TldrawUiButtonIcon, { icon, style })
         }
       ) });
     }
